src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -258,83 +193,34 @@ private IEnumerable<ITypeSymbol> InferTypeInArrowExpressionClause(ArrowExpressio
                 return SpecializedCollections.EmptyEnumerable<ITypeSymbol>();
             }
 
-            private IEnumerable<ITypeSymbol> InferTypesWorker(int position)
+            protected override IEnumerable<ITypeSymbol> InferTypesWorker_DoNotCallDirectly(int position)
             {
-                var syntaxTree = _semanticModel.SyntaxTree;
-                var token = syntaxTree.FindTokenOnLeftOfPosition(position, _cancellationToken);
+                var syntaxTree = SemanticModel.SyntaxTree;
+                var token = syntaxTree.FindTokenOnLeftOfPosition(position, CancellationToken);
                 token = token.GetPreviousTokenIfTouchingWord(position);
-                var parent = token.Parent;
 
-                // A couple of broken code scenarios where the new keyword in objectcreationexpression
-                // appears to be a part of a subsequent assignment.
-                if (token.Kind() == SyntaxKind.NewKeyword &&
```

Comment:
For consistency, these functions should only get the parent/left node and typeswitch.  Complex logic should be in the function that is dispatched to.